src/components/layout/ChatArea.tsx

```diff
@@ -20,6 +20,7 @@ import type { Message as MessageType, User } from "../../types";
 import EmojiSelector from "../ui/EmojiSelector";
 
 const EMPTY_ARRAY: User[] = [];
+let lastTypingTime = 0;
```

Comment:
Please dont

https://stackoverflow.com/questions/69719307/why-use-react-useref-rather-than-a-module-level-let-to-keep-track-of-a-variabl
https://hygraph.com/blog/react-useref-a-complete-guide

TLDR: Make something that is unique to each component instance. Global vars not good.

Reply:
Idk what happens when you go to another channel though. This seems tricky.

Reply:
We agreed this is fine for now